src/tribler/gui/tests/test_gui.py

```diff
@@ -76,6 +76,11 @@ def fixture_window(tmp_path_factory):
 
     window.close_tribler()
     screenshot(window, name="tribler_closing")
+
+    # Before quitting the application, wait for max 10 seconds until the core process has successfully finished
+    # Otherwise, process exits with non-zero exit code.
+    # See: https://github.com/Tribler/tribler/issues/7500
+    wait_for_signal(window.core_manager.core_process.finished, timeout=10)
```

Comment:
Question: are ten seconds always enough here? Something like 30 seconds looks safer to me

Reply:
In my opinion, 10 seconds is sufficient since the test is running with the test state directory which shouldn't be big. If the test is timing out for 10 seconds timeout value, then I think we should first investigate and then decide whether to increase the timeout.